lib/experimental/fonts-library/class-wp-fonts-library.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+class WP_Fonts_Library {
+
+	const ALLOWED_FONT_MIME_TYPES = array(
+		'otf'   => 'font/otf',
+		'ttf'   => 'font/ttf',
+		'woff'  => 'font/woff',
+		'woff2' => 'font/woff2',
+	);
+
+	/**
+	 * Returns the absolute path to the fonts directory.
+	 *
+	 * @return string Path to the fonts directory.
+	 */
+	static public function get_fonts_directory() {
+		return path_join( WP_CONTENT_DIR, 'fonts' );
+	}
+
+	/**
+	 * Returns the relative path to the fonts directory.
+	 *
+	 * @return string Relative path to the fonts directory.
+	 */
+	static public function get_relative_fonts_path() {
+		return content_url( '/fonts/' );
+	}
+
+	/**
+	 * Define WP_FONTS_DIR constant to make it available to the rest of the code.
+	 */
+	static public function define_fonts_directory() {
+		define( 'WP_FONTS_DIR', self::get_fonts_directory() );
+	}
+
+    /**
+     * Create fonts directory if it doesn't exist.
+     */
+	static public function create_fonts_directory() {
+		wp_mkdir_p( self::get_fonts_directory() );
+	}
+
+}
+
+add_action( 'init', array( 'WP_Fonts_Library', 'define_fonts_directory' ) );
+add_action( 'init', array( 'WP_Fonts_Library', 'create_fonts_directory' ) );
```

Comment:
This file should only contain the `class`. Move these hooks to  either `load.php` or to a new `lib/experimental/fonts-library/fonts-library.php` file. When introduced into Core, these 2 hooks will be contained elsewhere.

Reply:
I added a file https://github.com/WordPress/gutenberg/blob/afc6a48bd27e232e893e0a46e2bf0f2fc324d484/lib/experimental/fonts-library/fonts-library.php

Does it look good?